src/controller/CHIPDeviceController.cpp

```diff
@@ -1733,6 +1733,45 @@ void DeviceCommissioner::OnDone(app::ReadClient *)
             return CHIP_NO_ERROR;
         }
     });
+
+    err = mAttributeCache->ForEachAttribute(
+        app::Clusters::OperationalCredentials::Id, [this, &info](const app::ConcreteAttributePath & path) {
+            if (path.mAttributeId != app::Clusters::OperationalCredentials::Attributes::Fabrics::Id)
+            {
+                // Continue on
+                return CHIP_NO_ERROR;
+            }
+
+            switch (path.mAttributeId)
+            {
+            case app::Clusters::OperationalCredentials::Attributes::Fabrics::Id: {
+                app::Clusters::OperationalCredentials::Attributes::Fabrics::TypeInfo::DecodableType fabrics;
+                ReturnErrorOnFailure(
+                    this->mAttributeCache->Get<app::Clusters::OperationalCredentials::Attributes::Fabrics::TypeInfo>(path,
+                                                                                                                     fabrics));
+                auto iter = fabrics.begin();
+                while (iter.Next())
+                {
+                    auto & fabricDescriptor = iter.GetValue();
+                    ChipLogProgress(AppServer,
+                                    "DeviceCommissioner::OnDone - fabric.vendorId=0x%04X fabric.fabricId=0x" ChipLogFormatX64
+                                    " fabric.nodeId=0x" ChipLogFormatX64,
+                                    fabricDescriptor.vendorId, ChipLogValueX64(fabricDescriptor.fabricId),
```

Comment:
What is all this code for? It's really not clear what is attempted here, and for what. Also, this expects synchronous-like operation, and in common code.

Reply:
added comment explaining that this is checking for commissioner fabric already on device, and saving the nodeId